folder.go

```diff
@@ -73,6 +73,20 @@ func (f Folder) CreateDatacenter(datacenter string) (*Datacenter, error) {
 	return nil, err
 }
 
+func (f Folder) CreateFolder(name string) (*Folder, error) {
+	req := types.CreateFolder{
+		This: f.Reference(),
+		Name: name,
+	}
+
+	res, err := methods.CreateFolder(f.c, &req)
+	if res != nil {
```

Comment:
Hi @gavrie , thanks for the patch and CLA!  Could you change this to:
```go
if err != nil {
   return err
}
```
And at the end of the func: return NewFolder(...), nil
We use this same pattern in the rest of the code, except Folder.CreateDatacenter (which we should fix too).